import json
import logging
import os

from biothings.web.handlers import MetadataSourceHandler
from tornado.web import RequestHandler


class WebAppHandler(RequestHandler):
    def get(self):
        if self.render("dist/index.html"):
            self.render("dist/index.html")
        else:
            logging.info("Unable to find dist folder from react app.")


class NDESourceHandler(MetadataSourceHandler):
    """
    GET /v1/metadata
    """

    def load_from_cache(self, datasource):
        file_name = f"cache_{datasource}.json"
        folder = "metadata_completeness"
        cache_file = os.path.join(folder, file_name)
        if os.path.exists(cache_file):
            with open(cache_file, "r") as f:
                averages = json.load(f)
            return averages
        return None

    def calculate_metadata_compatibility_average(self, datasource):
        cached_averages = self.load_from_cache(datasource)
        if cached_averages is not None:
            return cached_averages

    def extras(self, _meta):
        source_info = {
            # Example Object
            # <SOURCE_NAME> = https://api.data.niaid.nih.gov/v1/metadata
            # <SOURCE_NAME> = {
            #       'name': 'A proper source name',
            #       'description': 'A short description of what the source offers, usually found on the source's about page',
            #       'schema': 'A dict where the key is their metadata variable and the value is our transformation. Ex: {"summary":"description"},
            #       'url': 'The source's URL',
            #       'identifier':'includedInDataCatalog.name',
            # }
            "veupath_collections": {
                "name": "VEuPath Collections",
                "abstract": "VEuPathCollections are NIAID supported IID component repositories of VEuPathDB which include collections of genomic data.",
                "description": "VEuPath Collections is the collection of component websites of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) that have been constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). \nThe VEuPath Collections include the following component websites as individual collections: \nAmoebaDB \nCryptoDB \nGiardiaDB \nHostDB \nPlasmoDB \nVectorBase \nFungiDB \nMicrosporidiaDB \nToxoDB \nTrichDB \nTriTrypDB \nPiroplasmaDB.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://veupathdb.org/veupathdb/app/",
                "identifier": "VEuPathDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
            },
            "amoebadb": {
                "name": "AmoebaDB",
                "abstract": "AmoebaDB is a NIAID supported IID repository that includes genomic data.",
                "description": "AmoebaDB is a component website of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) and is constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK).These resources stem from support initially provided for the Plasmodium Genome Database by the Burroughs Wellcome Fund (2000-2) and a research grant from NIAID (2002-6). The BRC program was initiated in 2004 to provide public access to computational platforms and analysis tools enabling collection, management, integration and mining of genomic information and other large-scale datasets relevant to infectious disease pathogens including their interaction with mammalian hosts and invertebrate vectors of disease.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://amoebadb.org/amoeba/app",
                "identifier": "AmoebaDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
                "parentCollection": {"id": "veupathdb"},
            },
            "cryptodb": {
                "name": "CryptoDB",
                "abstract": "CryptoDB is a NIAID supported IID repository that includes genomic data.",
                "description": "CryptoDB is a component website of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) and is constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). These resources stem from support initially provided for the Plasmodium Genome Database by the Burroughs Wellcome Fund (2000-2) and a research grant from NIAID (2002-6). The BRC program was initiated in 2004 to provide public access to computational platforms and analysis tools enabling collection, management, integration and mining of genomic information and other large-scale datasets relevant to infectious disease pathogens including their interaction with mammalian hosts and invertebrate vectors of disease.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://cryptodb.org/cryptodb/app",
                "identifier": "CryptoDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
                "parentCollection": {"id": "veupathdb"},
            },
            "giardiadb": {
                "name": "GiardiaDB",
                "abstract": "GiardiaDB is a NIAID supported IID repository that includes genomic data.",
                "description": "GiardiaDB is a component website of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) and is constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). These resources stem from support initially provided for the Plasmodium Genome Database by the Burroughs Wellcome Fund (2000-2) and a research grant from NIAID (2002-6). The BRC program was initiated in 2004 to provide public access to computational platforms and analysis tools enabling collection, management, integration and mining of genomic information and other large-scale datasets relevant to infectious disease pathogens including their interaction with mammalian hosts and invertebrate vectors of disease.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://giardiadb.org/giardiadb/app",
                "identifier": "GiardiaDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
                "parentCollection": {"id": "veupathdb"},
            },
            "hostdb": {
                "name": "HostDB",
                "abstract": "HostDB is a NIAID supported IID repository that includes genomic data.",
                "description": "HostDB is a component website of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) and is constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). These resources stem from support initially provided for the Plasmodium Genome Database by the Burroughs Wellcome Fund (2000-2) and a research grant from NIAID (2002-6). The BRC program was initiated in 2004 to provide public access to computational platforms and analysis tools enabling collection, management, integration and mining of genomic information and other large-scale datasets relevant to infectious disease pathogens including their interaction with mammalian hosts and invertebrate vectors of disease.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://hostdb.org/hostdb/app",
                "identifier": "HostDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
                "parentCollection": {"id": "veupathdb"},
            },
            "plasmodb": {
                "name": "PlasmoDB",
                "abstract": "PlasmoDB is a NIAID supported IID repository that includes genomic data.",
                "description": "PlasmoDB is a component website of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) and is constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). These resources stem from support initially provided for the Plasmodium Genome Database by the Burroughs Wellcome Fund (2000-2) and a research grant from NIAID (2002-6). The BRC program was initiated in 2004 to provide public access to computational platforms and analysis tools enabling collection, management, integration and mining of genomic information and other large-scale datasets relevant to infectious disease pathogens including their interaction with mammalian hosts and invertebrate vectors of disease.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://plasmodb.org/plasmo/app",
                "identifier": "PlasmoDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
                "parentCollection": {"id": "veupathdb"},
            },
            "vectorbase": {
                "name": "VectorBase",
                "abstract": "VectorBase is a NIAID supported IID repository that includes genomic data.",
                "description": "VectorBase is a component website of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) and is constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). These resources stem from support initially provided for the Plasmodium Genome Database by the Burroughs Wellcome Fund (2000-2) and a research grant from NIAID (2002-6). The BRC program was initiated in 2004 to provide public access to computational platforms and analysis tools enabling collection, management, integration and mining of genomic information and other large-scale datasets relevant to infectious disease pathogens including their interaction with mammalian hosts and invertebrate vectors of disease.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://vectorbase.org/vectorbase/app",
                "identifier": "VectorBase",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
                "parentCollection": {"id": "veupathdb"},
            },
            "fungidb": {
                "name": "FungiDB",
                "abstract": "FungiDB is a NIAID supported IID repository that includes genomic data.",
                "description": "FungiDB is a component website of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) and is constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). These resources stem from support initially provided for the Plasmodium Genome Database by the Burroughs Wellcome Fund (2000-2) and a research grant from NIAID (2002-6). The BRC program was initiated in 2004 to provide public access to computational platforms and analysis tools enabling collection, management, integration and mining of genomic information and other large-scale datasets relevant to infectious disease pathogens including their interaction with mammalian hosts and invertebrate vectors of disease.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://fungidb.org/fungidb/app",
                "identifier": "FungiDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
                "parentCollection": {"id": "veupathdb"},
            },
            "microsporidiadb": {
                "name": "MicrosporidiaDB",
                "abstract": "MicrosporidiaDB is a NIAID supported IID repository that includes genomic data.",
                "description": "MicrosporidiaDB is a component website of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) and is constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). These resources stem from support initially provided for the Plasmodium Genome Database by the Burroughs Wellcome Fund (2000-2) and a research grant from NIAID (2002-6). The BRC program was initiated in 2004 to provide public access to computational platforms and analysis tools enabling collection, management, integration and mining of genomic information and other large-scale datasets relevant to infectious disease pathogens including their interaction with mammalian hosts and invertebrate vectors of disease.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://microsporidiadb.org/micro/app",
                "identifier": "MicrosporidiaDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
                "parentCollection": {"id": "veupathdb"},
            },
            "toxodb": {
                "name": "ToxoDB",
                "abstract": "ToxoDB is a NIAID supported IID repository that includes genomic data.",
                "description": "ToxoDB is a component website of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) and is constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). These resources stem from support initially provided for the Plasmodium Genome Database by the Burroughs Wellcome Fund (2000-2) and a research grant from NIAID (2002-6). The BRC program was initiated in 2004 to provide public access to computational platforms and analysis tools enabling collection, management, integration and mining of genomic information and other large-scale datasets relevant to infectious disease pathogens including their interaction with mammalian hosts and invertebrate vectors of disease.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://toxodb.org/toxo/app",
                "identifier": "ToxoDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
                "parentCollection": {"id": "veupathdb"},
            },
            "trichdb": {
                "name": "TrichDB",
                "abstract": "TrichDB is a NIAID supported IID repository that includes genomic data.",
                "description": "TrichDB is a component website of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) and is constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). These resources stem from support initially provided for the Plasmodium Genome Database by the Burroughs Wellcome Fund (2000-2) and a research grant from NIAID (2002-6). The BRC program was initiated in 2004 to provide public access to computational platforms and analysis tools enabling collection, management, integration and mining of genomic information and other large-scale datasets relevant to infectious disease pathogens including their interaction with mammalian hosts and invertebrate vectors of disease.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://trichdb.org/trichdb/app",
                "identifier": "TrichDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
                "parentCollection": {"id": "veupathdb"},
            },
            "tritrypdb": {
                "name": "TriTrypDB",
                "abstract": "TriTrypDB is a NIAID supported IID repository that includes genomic data.",
                "description": "TriTrypDB is a component website of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) and is constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). These resources stem from support initially provided for the Plasmodium Genome Database by the Burroughs Wellcome Fund (2000-2) and a research grant from NIAID (2002-6). The BRC program was initiated in 2004 to provide public access to computational platforms and analysis tools enabling collection, management, integration and mining of genomic information and other large-scale datasets relevant to infectious disease pathogens including their interaction with mammalian hosts and invertebrate vectors of disease.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://tritrypdb.org/tritrypdb/app",
                "identifier": "TriTrypDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
                "parentCollection": {"id": "veupathdb"},
            },
            "piroplasmadb": {
                "name": "PiroplasmaDB",
                "abstract": "PiroplasmaDB is a NIAID supported IID repository that includes genomic data.",
                "description": "PiroplasmaDB is a component website of the Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) and is constructed using the same infrastructure, data analysis, and loading procedures. VEuPathDB is is one of two bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). These resources stem from support initially provided for the Plasmodium Genome Database by the Burroughs Wellcome Fund (2000-2) and a research grant from NIAID (2002-6). The BRC program was initiated in 2004 to provide public access to computational platforms and analysis tools enabling collection, management, integration and mining of genomic information and other large-scale datasets relevant to infectious disease pathogens including their interaction with mammalian hosts and invertebrate vectors of disease.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://piroplasmadb.org/piro/app",
                "identifier": "PiroplasmaDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
                "parentCollection": {"id": "veupathdb"},
            },
            "massive": {
                "name": "MassIVE",
                "abstract": "MassIVE is a NIH supported Basic science repository that includes mass spectrometry data.",
                "description": "MassIVE is a community resource developed by the NIH-funded Center for Computational Mass Spectrometry to promote the global, free exchange of mass spectrometry data. MassIVE datasets can be assigned ProteomeXchange accessions to satisfy publication requirements.",
                "schema": {
                    "dataset": "identifier",
                    "task": "url",
                    "repo_path": "distribution",
                    "title": "name",
                    "site": "sdPublisher",
                    "description": "description",
                    "keywords": "keywords",
                    "create_time": "dateCreated",
                    "instrument_resolved": "measurementTechnique",
                    "species_resolved": "species",
                    "pis": "author",
                    "publications": "citation",
                    "privacy": "conditionsOfAccess",
                },
                "url": "https://massive.ucsd.edu/ProteoSAFe/static/massive.jsp",
                "identifier": "MassIVE",
                "conditionsOfAccess": "Open",
                "genre": "Generalist",
            },
            "malariagen": {
                "name": "MalariaGEN",
                "abstract": "Malaria Genomic Epidemiology Network (MalariaGEN) is a IID repository that includes clinical data.",
                "description": "MalariaGEN is an international network of researchers focused on understanding the genetic variations in humans, Plasmodium parasites, and Anopheles mosquitoes to better control malaria transmission. By leveraging advanced genomic tools and techniques, MalariaGEN aims to uncover how genetic differences influence susceptibility to malaria, track insecticide resistance in mosquitoes, and detect drug resistance in parasites. The network's efforts are crucial for developing new interventions and maintaining the effectiveness of existing ones. Coordinated from the Wellcome Sanger Institute in the UK, MalariaGEN collaborates with over 200 partners across more than 40 countries, primarily in malaria-endemic regions.",
                "schema": {
                    "title": "name",
                    "released on": "datePublished",
                    "project": "isPartOf",
                    "url": "identifier",
                    "keywords": "keywords",
                    "datasets": "distribution",
                    "citations": "citation",
                    "data package contact": "author",
                    "spatialCoverage": "spatialCoverage",
                },
                "url": "https://www.malariagen.net/",
                "identifier": "MalariaGEN",
                "conditionsOfAccess": "Varied",
                "genre": "IID",
            },
            "dryad": {
                "name": "Dryad Digital Repository",
                "abstract": "Dryad Digital Repository is a GREI repository that includes most data types and domains.",
                "description": "Dryad is an open source, community driven project that takes a unique approach to data publication and digital preservation. Dryad focuses on search, presentation, and discovery and delegates the responsibility for the data preservation function to the underlying repository with which it is integrated. Dryad aims to allow researchers to validate published findings, explore new analysis methodologies, re-purpose data for research questions unanticipated by the original authors, and perform synthetic studies such as formal meta-analyses.",
                "schema": {
                    "name": "name",
                    "description": "description",
                    "contentUrl": "contentUrl",
                    "identifier": "identifier, doi",
                    "keywords": "keywords",
                    "creator": "author",
                    "distribution": "distribution",
                    "temporalCoverage": "temporalCoverage",
                    "spatialCoverage": "spatialCoverage",
                    "citation": "citation",
                    "license": "license",
                    "datePublished": "datePublished",
                    "conditions": "healthCondition.name",
                    "outcomeNames": "variable_measured",
                    "digitalObjectId": "doi",
                    "studyMetadataDoi": "doi",
                    "extractedBriefSummary": "description",
                    "draftCreatedDate": "dateCreated",
                    "postedDate": "datePublished",
                    "updatedDate": "dateModified",
                },
                "url": "https://datadryad.org",
                "identifier": "Dryad Digital Repository",
                "conditionsOfAccess": "Open",
                "genre": "Generalist",
            },
            "vivli": {
                "name": "Vivli",
                "abstract": "Vivli is a GREI repository that includes most data types and domains.",
                "description": "Vivli is an independent non-profit organization launched in 2016. Vivli evolved from a project of The Multi-Regional Clinical Trials Center of Brigham and Women’s Hospital and Harvard(MRCT Center) to enhance access to clinical trials data by promoting data sharing and transparency. In 2013, the MRCT Center and a diverse group of global stakeholders embarked on a mission to define, design, and launch an innovative platform solution for global clinical trial data sharing. The Vivli platform provides access to anonymized individual participant-level data(IPD) or the raw data that is collected during a clinical trial. The clinical trials represented in Vivli are global and contributed by a diverse group of data contributors. By serving as a global trusted platform, Vivli increases the discoverability of available data in the wider research ecosystem, and increases the overall capacity worldwide for effective data sharing, aggregation, re-use, and novel analysis of valuable clinical research data to advance science and improve public health. The Vivli platform is a cloud-based platform that consists of a dynamic search engine, a data repository, and a secure research environment.",
                "schema": {
                    "nctId": "identifer",
                    "secondaryIds": "identifier",
                    "registryInfo": "sdPublishers",
                    "principalInvestigator": "author",
                    "studyTitle": "name",
                    "leadSponsor": "funding",
                    "collaborators": "funding",
                    "studyStartDate": "temporalInterval.startDate",
                    "actualStudyCompletionDate": "temporalInterval.endDate",
                    "locationsOfStudySites": "spatialCoverage",
                    "phase": "keywords",
                    "studyType": "keywords",
                    "conditions": "healthCondition.name",
                    "outcomeNames": "variable_measured",
                    "digitalObjectId": "doi",
                    "studyMetadataDoi": "doi",
                    "extractedBriefSummary": "description",
                    "draftCreatedDate": "dateCreated",
                    "postedDate": "datePublished",
                    "updatedDate": "dateModified",
                },
                "url": "https://vivli.org/",
                "identifier": "Vivli",
                "conditionsOfAccess": "Restricted",
                "genre": "Generalist",
            },
            "ncbi_pmc": {
                "name": "NCBI PMC",
                "description": "PubMed Central® (PMC) is a free full-text archive of biomedical and life sciences journal literature at the U.S. National Institutes of Health's National Library of Medicine (NIH/NLM). In keeping with NLM's legislative mandate to collect and preserve the biomedical literature, PMC is part of the NLM collection, which also includes NLM's extensive print and licensed electronic journal holdings and supports contemporary biomedical and health care research and practice as well as future scholarship. Available to the public online since 2000, PMC was developed and is maintained by the National Center for Biotechnology Information(NCBI) at NLM.",
                "schema": {
                    "article-id": "identifer",
                    "article-id": "url",
                    "volume": "citation.volume",
                    "journal-title": "citation.journalName",
                    "pub-date": "citation.datePublished",
                    "article-title": "citation.name",
                    "article-id": "citation.pmid",
                    "article-id": "citation.identifier",
                    "article-id": "citation.url",
                    "article-id": "citation.doi",
                    "abstract": "description",
                    "contrib": "author",
                },
                "url": "https://www.ncbi.nlm.nih.gov/pmc/",
                "identifier": "NCBI PMC",
                "conditionsOfAccess": "Unknown",
                "genre": "Generalist",
            },
            "veupathdb": {
                "name": "VEuPathDB",
                "abstract": "The Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) is a NIAID supported IID repository that includes genomic and phenotypic data for eukaryotic pathogens and invertebrate vectors of infectious disease.",
                "description": "The Eukaryotic Pathogen, Vector and Host Informatics Resource (VEuPathDB) is one of two Bioinformatics Resource Centers (BRCs) funded by the US National Institute of Allergy and Infectious Diseases (NIAID), with additional support from the Wellcome Trust (UK). VEuPathDB provides access to diverse genomic and other large scale datasets related to eukaryotic pathogens and invertebrate vectors of disease. Organisms supported by this resource include (but are not limited to) the NIAID list of emerging and re-emerging infectious diseases.",
                "schema": {
                    "id": "identifer",
                    "displayName": "name",
                    "contact_name": "author",
                    "summary": "description",
                    "type": "measurementTechnique",
                    "sdPublisher": "project_id",
                    "short_attribution": "creditText",
                    "release_policy": "conditionOfAccess",
                    "version": "dateModified",
                    "author": "affiliation",
                    "GenomeHistory": "dateUpdated",
                    "Version": "datePublished",
                    "organism": "species",
                    "HyperLinks": "distribution",
                    "gene_count": "variableMeasured",
                    "gene_type": "GeneTypeCounts",
                },
                "url": "https://veupathdb.org/veupathdb/app/",
                "identifier": "VEuPathDB",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
            },
            "acd_niaid": {
                "name": "AccessClinicalData@NIAID",
                "abstract": "AccessClinicalData is a NIAID supported IID repository that includes clinical trials data.",
                "description": "AccessClinicalData@NIAID is a NIAID cloud-based, secure data platform that enables sharing of and access to reports and data sets from NIAID COVID-19 and other sponsored clinical trials for the basic and clinical research community.",
                "schema": {
                    "title": "name",
                    "cmc_unique_id": "identifier",
                    "brief_summary": "description",
                    "data_availability_date": "datePublished",
                    "most_recent_update": "dateModified",
                    "data_available": "additionalType",
                    "creator": "funding.funder.name",
                    "nct_number": "nctid, identifier",
                    "condition": "healthCondition",
                    "clinical_trial_website": "mainEntityOfPage",
                    "publications": "citation",
                    "data_available_for_request": "conditionsOfAccess",
                },
                "url": "https://accessclinicaldata.niaid.nih.gov/",
                "identifier": "AccessClinicalData@NIAID",
                "conditionsOfAccess": "Varied",
                "genre": "IID",
            },
            "sb_apps": {
                "name": "Seven Bridges Public Apps Gallery",
                "description": "The Seven Bridges Public Apps Gallery offers a repository of publicly available apps suitable for many different types of data analysis. Apps include both tools (individual bioinformatics utilities) and workflows (chains or pipelines of connected tools). The publicly available apps are maintained by the Seven Bridges Platform bioinformatics team to represent the latest tool versions.",
                "schema": {
                    "class": "applicationCategory",
                    "label": "name",
                    "description": "description",
                    "inputs": "input",
                    "outputs": "output",
                    "requirements": "softwareRequirements",
                    "sbg:image_url": "thumbnailUrl",
                    "sbg:toolkit": "applicationSuite",
                    "sbg:license": "license",
                    "sbg:links": "codeRepository",
                    "sbg:categories": "applicationSubCategory",
                    "sbg:toolAuthor": "author",
                    "sbg:appVersion": "softwareVersion",
                    "sbg:id": "url, identifier",
                    "sbg:revisionNotes": "version",
                    "sbg:modifiedOn": "dateModified",
                    "sbg:createdOn": "dateCreated",
                    "sbg:contributors": "contributor",
                    "sbg:publisher": "sdPublisher",
                    "sbg:workflowLanguage": "programmingLanguage",
                },
                "url": "https://workspace.niaiddata.org/public/apps",
                "identifier": "PublicApps@SevenBridges",
                "conditionsOfAccess": "Unknown",
                "genre": "Generalist",
            },
            "zenodo": {
                "name": "Zenodo",
                "abstract": "Zenodo is a GREI repository that includes most data types and domains.",
                "description": "The OpenAIRE project, in the vanguard of the open access and open data movements in Europe was commissioned by the EC to support their nascent Open Data policy by providing a catch-all repository for EC funded research. CERN, an OpenAIRE partner and pioneer in open source, open access and open data, provided this capability and Zenodo was launched in May 2013.",
                "schema": {
                    "title": "name",
                    "header": "dateModifed, url",
                    "description": "description",
                    "date": "datePublished",
                    "language": "inLanguage.name",
                    "subject": "keywords",
                    "relatedIdentifier": "topicCategory",
                    "xml[DOI]": "doi",
                    "xml[resourceTypeGeneral]": "@type",
                    "xml[creator][creatorName]": "author.name",
                    "xml[creator][affiliation]": "author.affiliation",
                    "xml[creator][nameIdentifier]": "author.identifier",
                    "xml[rights]": "conditionsOfAccess, license",
                    "xml[relatedIdentifier][relationType]": "citedBy",
                    "xml[contributor][contributorName]": "funding.funder.name",
                },
                "url": "https://zenodo.org/",
                "identifier": "Zenodo",
                "conditionsOfAccess": "Varied",
                "genre": "Generalist",
            },
            "dde": {
                "name": "Data Discovery Engine",
                "abstract": "The Data Discovery Engine is a NIAID supported metadata registry that includes metadata from IID and general repositories.",
                "description": "The Data Discovery Engine is a streamlined process to create, distribute and harves findable metadata via interoperable Schema.org schemas. The biomedical and informatics communities have largely endorsed the spirit and basic components of the FAIR Data Principles. Biomedical data producers, including CTSA hubs, need actionable best-practice guidance on how to make their data discoverable and reusable, and bring the practical benefits of data sharing to researcher's own research projects, as well as the research community as a whole.",
                "schema": {
                    "creator": "author",
                    "_id": "identifier",
                    "date_created": "dateCreated",
                    "last_updated": "dateModifed",
                    "@type": "@type",
                    "measurementTechnique": "measurementTechnique",
                    "infectiousAgent": "infectiousAgent",
                    "infectiousDisease": "infectiousDisease",
                    "species": "species",
                },
                "url": "https://discovery.biothings.io/",
                "identifier": "Data Discovery Engine",
                "conditionsOfAccess": "Varied",
                "genre": "Generalist",
            },
            "ncbi_geo": {
                "name": "NCBI GEO",
                "abstract": "Gene Expression Omnibus (GEO) is an NIH supported repository that includes microarray and next-generation sequencing data for high-throughput functional genomics for most domains.",
                "description": "GEO is a public functional genomics data repository supporting MIAME-compliant data submissions. Array- and sequence-based data are accepted. Tools are provided to help users query and download experiments and curated gene expression profiles.",
                "schema": {
                    "_id": "identifier",
                    "contributor(s)": "author",
                    "organization": "publisher",
                    "title": "name",
                    "organism": "species",
                    "experiment type": "measurementTechnique",
                    "summary": "description",
                    "submission date": "datePublished",
                    "last update date": "dateModified",
                    "citation(s)": "citation",
                },
                "url": "https://www.ncbi.nlm.nih.gov/geo/",
                "identifier": "NCBI GEO",
                "conditionsOfAccess": "Unknown",
                "genre": "Generalist",
            },
            "immport": {
                "name": "ImmPort",
                "abstract": "The Immunology Database and Analysis Portal (ImmPort) is a NIAID supported IID repository that includes multimodal immunological data.",
                "description": "The ImmPort project provides advanced information technology support in the archiving and exchange of scientific data for the diverse community of life science researchers supported by NIAID/DAIT and serves as a long-term, sustainable archive of research and clinical data. The core component of ImmPort is an extensive data warehouse containing experimental data and metadata describing the purpose of the study and the methods of data generation. The functionality of ImmPort will be expanded continuously over the life of the BISC project to accommodate the needs of expanding research communities. The shared research and clinical data, as well as the analytical tools in ImmPort are available to any researcher after registration.",
                "schema": {
                    "_id": "identifer",
                    "creator": "author",
                    "citations": "citedBy",
                    "identifers": "identifer",
                    "species": "species",
                    "measurementTechnique": "measurementTechnique",
                    "distribution": "distribution",
                    "includedInDataCatalog": "includedInDataCatalog",
                    "date": "date",
                },
                "url": "https://www.immport.org/shared/home",
                "identifier": "ImmPort",
                "conditionsOfAccess": "Closed",
                "genre": "IID",
            },
            "omicsdi": {
                "name": "OmicsDI",
                "abstract": "Omics Discovery Index (OmicsDI) is a generalist repository that is part of ELIXIR infrastructure that includes multi-omics data for most domains.",
                "description": "The Omics Discovery Index (OmicsDI) provides a knowledge discovery framework across heterogeneous omics data (genomics, proteomics, transcriptomics and metabolomics).",
                "schema": {
                    "_id": "identifer",
                    "citation": "citation",
                    "creator": "author",
                    "description": "description",
                    "distribution": "distribution",
                    "keywords": "keywords",
                    "name": "name",
                    "sameAs": "sameAs",
                    "variableMeasured": "variableMeasured",
                },
                "url": "https://www.omicsdi.org/",
                "identifier": "Omics Discovery Index (OmicsDI)",
                "conditionsOfAccess": "Unknown",
                "genre": "Generalist",
            },
            "mendeley": {
                "name": "Mendeley Data",
                "abstract": "Mendeley Data is a GREI repository that includes most data types and domains.",
                "description": "Mendeley Data, a product of Elsevier, is one of the newest entrants in the research data repository landscape; the platform was released in April 2016. Mendeley Data is a general-purpose repository, allowing researchers in any field to upload and publish research data. Mendeley Data also allows researchers to share unpublished data privately with research collaborators.",
                "schema": {
                    "id": "identifer",
                    "doi": "doi",
                    "name": "name",
                    "description": "description",
                    "contributors": "contributors",
                    "files": "distribution",
                    "articles": "citation",
                    "categories": "keywords",
                    "publish_date": "datePublished",
                    "related_links": "citation",
                    "modified_on": "dateModified",
                    "links": "url",
                    "repository": "sdPublisher",
                },
                "url": "https://data.mendeley.com/",
                "identifier": "Mendeley",
                "conditionsOfAccess": "Varied",
                "genre": "Generalist",
            },
            "reframedb": {
                "name": "ReframeDB",
                "abstract": "ReframeDB is a Calibr maintained repository that has a screening set of 12,000 compounds for drug repositioning.",
                "description": "The ReFRAME collection of 12,000 compounds is a best-in-class drug repurposing library containing nearly all small molecules that have reached clinical development or undergone significant preclinical profiling. The purpose of such a screening collection is to enable rapid testing of compounds with demonstrated safety profiles in new indications, such as neglected or rare diseases, where there is less commercial motivation for expensive research and development.",
                "schema": {
                    "assay_id": "identifer",
                    "assay_title": "name",
                    "title_short": "alternateName",
                    "authors": "author",
                    "summary": "description",
                    "purpose": "description",
                    "protocol": "description",
                    "readout": "description",
                    "detection_method": "description",
                    "detection_reagents": "description",
                    "components": "description",
                    "drug_conc": "description",
                    "indication": "healthCondition",
                    "assay_type": "measurementTechnique",
                    "bibliography": "citation",
                },
                "url": "https://reframedb.org/",
                "identifier": "ReframeDB",
                "conditionsOfAccess": "Restricted",
                "genre": "IID",
            },
            "clinepidb": {
                "abstract": "ClinEpiDB is a NIAID supported IID repository that includes epidemiological data.",
                "description": "ClinEpiDB, launched in February 2018, is an open-access exploratory data analysis platform. We integrate data from high quality epidemiological studies, and offer tools and visualizations to explore the data within the browser in a point and click interface. We enable investigators to maximize the utility and reach of their data and to make optimal use of data released by others. ClinEpiDB is led by a team of scientists and developers based at the University of Pennsylvania, the University of Georgia, Imperial College London, and several other academic institutions. Currently, we are funded by the Bill and Melinda Gates Foundation for resource development and data integration, and by NIAID for integration of data from the International Centers of Excellence in Malaria Research (ICEMR).",
                "identifier": "ClinEpiDB",
                "name": "ClinEpiDB",
                "schema": {
                    "Study_Design": "measurementTechnique",
                    "description": "description",
                    "summary": "description",
                    "WHO": "keywords",
                    "Participant_Type": "keywords",
                    "Country": "spatialCoverage",
                    "study_access": "conditionOfAccess",
                    "Years": "temporalCoverage",
                    "disease": "healthCondition",
                    "Publications": "citation",
                    "release_date": "distribution.dateModified",
                    "dataset_name": "distribution.name",
                    "contact_name": "author.name",
                    "Contacts": "affiliation",
                    "HyperLinks": "isBasedOn.url",
                },
                "url": "https://clinepidb.org/ce/app",
                "conditionsOfAccess": "Varied",
                "genre": "IID",
            },
            "lincs": {
                "abstract": "The Library of Integrated Network-Based Cellular Signatures (LINCS) Data Portal is an NIH supported repository that includes gene expression and other cellular processes data.",
                "description": "The BD2K-LINCS DCIC is comprised of four major components: Integrated Knowledge Environment (IKE), Data Science Research (DSR), Community Training and Outreach (CTO) and Consortium Coordination and Administration (CCA). The Center is constructing a high-capacity scalable integrated knowledge environment enabling federated access, intuitive querying and integrative analysis and visualization across all LINCS resources and many additional external data types from other relevant resources. The Center’s data science research projects are aimed at addressing various data integration and intracellular molecular regulatory network challenges. The Center aims to develop: 1) methods to connect cellular and organismal phenotypes with molecular cellular signatures, and 2) novel data visualization methods for dynamically interacting with large-genomics and proteomics datasets.",
                "identifier": "LINCS",
                "name": "LINCS",
                "schema": {
                    "centerdatasetid": "url",
                    "funding": "funding",
                    "assayoverview": "description",
                    "centerurl": "author.url",
                    "description": "description",
                    "principalinvestigator": "author.name",
                    "centerfullname": "author.affiliation",
                    "datemodified": "dateUpdated",
                    "screeninglabinvestigator": "author.name",
                    "datasetname": "name",
                    "datasetid": "identifier",
                    "datereleased": "datePublished",
                    "assayname": "measurementTechnique",
                    "assayformat": "measurementTechnique",
                    "assaydesignmethod": "keywords",
                    "physicaldetection": "variableMeasured",
                    "biologicalprocess": "keywords",
                    "technologies": "keywords",
                    "biologicalbucket": "keywords",
                    "endpointcategorization": "keywords",
                    "size": "distribution.contentSize",
                    "datasetgroup": "isRelatedTo",
                    "toollink": "isBasedOn",
                    "tool": "isBasedOn",
                    "protocol": "isBasedOn.url",
                    "protein": "keywords",
                },
                "url": "https://lincsportal.ccs.miami.edu/",
                "conditionsOfAccess": "Unknown",
                "genre": "Generalist",
            },
            "dataverse": {
                "abstract": "Harvard Dataverse is a GREI repository that includes most data types and domains.",
                "description": "The Harvard Dataverse Repository is a free data repository open to all researchers from any discipline, both inside and outside of the Harvard community, where you can share, archive, cite, access, and explore research data. Each individual Dataverse collection is a customizable collection of datasets (or a virtual repository) for organizing, managing, and showcasing datasets.",
                "identifier": "Harvard Dataverse",
                "name": "Harvard Dataverse",
                "schema": {
                    "@context": "@context",
                    "@type": "@type",
                    "identifier": ["url", "doi", "identifier"],
                    "name": "name",
                    "creator": "author",
                    "author": "author",
                    "datePublished": "datePublished",
                    "dateModified": "dateModified",
                    "description": "description",
                    "keywords": "keywords",
                    "citation": "citation.citation",
                    "license": "license",
                    "includedInDataCatalog": "includedInDataCatalog",
                    "publisher": "sdPublisher",
                    "funder": "funder",
                    "distribution": "distribution",
                    "type": "@type",
                    "global_id": "doi",
                    "published_at": "datePublished",
                    "identifier_of_dataverse": "sdPublisher.identifier",
                    "name_of_dataverse": "sdPublisher.name",
                    "subjects": "topicCategory",
                    "createdAt": "dateCreated",
                    "updatedAt": "dateModified",
                    "authors": "author.name",
                },
                "url": "https://dataverse.harvard.edu/",
                "conditionsOfAccess": "Varied",
                "genre": "Generalist",
            },
            "hubmap": {
                "abstract": "The Human BioMolecular Atlas Program (HuBMAP) is an NIH supported repository that includes multimodal data of healthy cells in the human body.",
                "description": "HuBMAP is part of a rich ecosystem of established and emerging atlasing programs supported by NIH and globally by other funding organizations, many of which are focused on specific organs or diseases. HuBMAP has connected with these programs to ensure data interoperability, avoid duplication of work, and leverage and synergize gained knowledge. The consortium has organized a number of events to bring together these communities to discuss topics of shared interest and is committed to improving coordination and collaboration among different programs. In addition, many of the HuBMAP PIs had been or are still actively participating in these efforts, helping with cross-pollination and advancing our global understanding. HuBMAP, as its name implies, was specifically initiated to resolve the challenge of building integrated, comprehensive, high-resolution spatial maps of human tissues and organs, which has resulted in HuBMAP providing leadership in the ecosystem around techniques for integrating disparate, multi-dimensional and multi-scale datasets, the development of a Common Coordinate Framework (CCF) for integrating data across many individuals, and the development and validation of these assays. To further increase interoperability, HuBMAP has adopted a number of standards and processes developed by other domain expert consortia, working and is actively involved in the knowledge exchange. The consortium sees itself as an integral part of the ecosystem, sharing its strengths and actively contributing to the community.",
                "identifier": "HuBMAP",
                "name": "HuBMAP",
                "schema": {
                    "hubmap_id": "name",
                    "title": "description",
                    "anatomy_0": "keywords",
                    "anatomy_1": "keywords",
                    "anatomy_2": "keywords",
                    "display_subtype": "keywords",
                    "contacts.affiliation": "author.affiliation",
                    "contacts.first_name": "author.givenName",
                    "contacts.last_name": "author.familyName",
                    "contacts.middle_name_or_initial": "author.givenName",
                    "contacts.name": "author.name",
                    "contacts.orcid_id": "author.identifier",
                    "contributors.affiliation": "author.affiliation",
                    "contributors.first_name": "author.givenName",
                    "contributors.last_name": "author.familyName",
                    "contributors.middle_name_or_initial": "author.givenName",
                    "contributors.name": "author.name",
                    "contributors.orcid_id": "author.identifier",
                    "created_timestamp": "dateCreated",
                    "last_modified_timestamp": "dateModified",
                    "published_timestamp": "datePublished",
                    "data_access_level": "isAccessibleForFree",
                    "data_types": "measurementTechnique.name",
                    "dataset_info": "measurementTechnique.description",
                    "doi_url": "doi",
                    "uuid": "url",
                    "file.description": "distribution.name",
                    "file.mapped_description": "distribution.description",
                    "file.rel_path": "distribution.contentUrl",
                    "file.size": "distribution.contentSize",
                    "file.type": "distribution.encodingFormat",
                    "version": "version",
                    "metadata.dag_provenance_list": "isBasedOn.name",
                    "metadata.protocols_io_doi": "isBasedOn.doi",
                },
                "url": "https://hubmapconsortium.org/",
                "conditionsOfAccess": "Varied",
                "genre": "Generalist",
            },
            "ncbi_sra": {
                "abstract": "Sequence Read Archive (SRA) is the NIH supported largest publicly available repository of high throughput sequencing data that includes raw sequencing data and alignment information for most domains.",
                "description": "Sequence Read Archive(SRA) data, available through multiple cloud providers and NCBI servers, is the largest publicly available repository of high throughput sequencing data. The archive accepts data from all branches of life as well as metagenomic and environmental surveys. SRA stores raw sequencing data and alignment information to enhance reproducibility and facilitate new discoveries through data analysis.",
                "identifier": "NCBI SRA",
                "name": "NCBI SRA",
                "schema": {
                    "Accession": "url",
                    "Updated": "dateModified",
                    "Published": "datePublished",
                    "Recieved": "dateCreated",
                    "Visibility": "conditionsOfAccess",
                    "ReplacedBy": "sameAs",
                    "study_title": "name",
                    "study_abstract": "description",
                    "contact_name": "author",
                    "GCP_url": "contentUrl",
                    "GCP_free_egress": "isAccessibleForFree",
                    "AWS_url": "contentUrl",
                    "AWS_free_egress": "isAccessibleForFree",
                    "organism_taxid": "species.identifier",
                    "organism_name": "species.name",
                    "run_accession": "isBasedOn.identifier",
                    "BioProject": "isBasedOn.identifier",
                    "experiment_accession": "isBasedOn.identifier",
                    "sample_accession": "isBasedOn.identifier",
                    "instrument": "isBasedOn.identifier",
                    "cell line": "isBasedOn.identifier",
                    "HapMap sample ID": "isBasedOn.identifier",
                },
                "url": "https://www.ncbi.nlm.nih.gov/sra",
                "conditionsOfAccess": "Varied",
                "genre": "Generalist",
            },
            "vdj": {
                "abstract": "VDJServer is a NIAID supported repository that includes immune repertoire sequencing data.",
                "description": "VDJServer is a free, scalable resource for performing immune repertoire analysis and sharing data. VDJServer Community Data Portal is part of the AIRR Data Commons. Funded by a National Institute of Allergy and Infectious Diseases research grant (#1R01A1097403), the VDJServer project is led by The University of Texas Southwestern (UTSW) Medical Center in collaboration with the J. Craig Venter Institute and Yale University. The Texas Advanced Computing Center (TACC) at The University of Texas at Austin leads the cyberinfrastructure implementation, including the high performance computing (HPC) systems, storage, and software solutions.",
                "identifier": "VDJServer",
                "name": "VDJServer",
                "schema": {
                    "study_id": "identifier",
                    "study_title": "name",
                    "study_type.label": "measurementTechnique.name",
                    "study_type.id": "measurementTechnique.identifier",
                    "study_description": "description",
                    "inclusion_exclusion_criteria": "description",
                    "lab_name": "author.name",
                    "lab_address": "author.affiliation",
                    "submitted_by": "author.name",
                    "collected_by": "author.name",
                    "grants": "description",
                    "pub_ids": "citation",
                    "adc_publish_date": "datePublished",
                    "adc_update_date": "dateModified",
                    "archive_file": "distribution.contentUrl",
                    "download_url": "distribution.contentUrl",
                    "file_size": "distribution.contentSize",
                },
                "url": "https://vdj-staging.tacc.utexas.edu/community/",
                "conditionsOfAccess": "Unknown",
                "genre": "IID",
            },
            "microbiomedb": {
                "abstract": "MicrobiomeDB is a NIAID supported repository that includes clinical microbiome data and analysis tools.",
                "description": "MicrobiomeDB was developed as a discovery tool that empowers researchers to fully leverage their experimental metadata to construct queries that interrogate microbiome datasets.",
                "name": "MicrobiomeDB",
                "identifier": "MicrobiomeDB",
                "schema": {
                    "displayName": "name",
                    "study_categories": "keywords",
                    "type": "keywords",
                    "description": "description",
                    "institution": "contributor.affiliation",
                    "contact": "contributor.name",
                    "study_access": "conditionsOfAccess",
                    "email": "contributor.email",
                    "card_points": "description",
                    "card_headline": "doi",
                    "summary": "abstract",
                    "dataset_id": "identifier",
                    "dataset_name": "alternateName",
                    "pmids_download": "citation, funding",
                    "contacts": "author",
                    "DownloadVersion": "distribution",
                    "Publications": "citation, funding",
                    "AssociatedDatasets": "isPartOf, isRelatedTo",
                    "HyperLinks": "url",
                },
                "url": "https://microbiomedb.org/mbio/app",
                "conditionsOfAccess": "Open",
                "genre": "IID",
            },
            "qiita": {
                "abstract": "Qiita is a repository that includes microbiome data and analysis tools.",
                "description": "Qiita(canonically pronounced cheetah) is an entirely open-source microbial study management platform. It allows users to keep track of multiple studies with multiple 'omics data. Additionally, Qiita is capable of supporting multiple analytical pipelines through a 3rd-party plugin system, allowing the user to have a single entry point for all of their analyses. Qiita provides database and compute resources to the global community, alleviating the technical burdens that are typically limiting for researchers studying microbial ecology(e.g. familiarity with the command line or access to compute power).Qiita's platform allows for quick reanalysis of the datasets that have been deposited using the latest analytical technologies. This means that Qiita's internal datasets are living data that is periodically re-annotated according to current best practices.",
                "identifier": "Qiita",
                "name": "Qiita",
                "schema": {
                    "study_abstract": "description",
                    "study_id": "url",
                    "study_alias": "_id",
                    "study_title": "name",
                    "study_tags": "keywords",
                    "pi": "author",
                    "pubs": "doi,pmids",
                    "ebi_study_accession": "mainEntityOfPage",
                },
                "url": "https://qiita.ucsd.edu/",
                "conditionsOfAccess": "Unknown",
                "genre": "IID",
            },
            "hca": {
                "abstract": "The Human Cell Atlas is a repository that includes multimodal data of cells in the human body.",
                "description": "Thanks to new single cell genomics and spatial imaging technologies developed since the late 2000s and early 2010s, it is now possible to measure gene expression profiles in individual cells. These large scale data can be used with machine learning algorithms to decipher how the cells differ from and interact with their neighbors, and how they form and function in the tissue. This now allows scientists to identify and understand cell types in unprecedented detail, resolution and breadth. The Human Cell Atlas (HCA) is an international group of researchers using a combination of these new technologies to create cellular reference maps with the position, function and characteristics of every cell type in the human body.",
                "identifier": "Human Cell Atlas",
                "name": "Human Cell Atlas",
                "schema": {
                    "entryId": "url, _id",
                    "projectTitle": "name",
                    "projectDescription": "description",
                    "institution": "author.affiliation",
                    "laboratory": "author.affiliation",
                    "contactName": "author.name",
                    "projectRole": "author.role",
                    "email": "author.email",
                    "publicationTitle": "ncitation.ame",
                    "publicationUrl": "citation.url",
                    "doi": "citation.doi",
                    "supplementaryLinks": "mainEntityOfPage, isBasedOn.codeRepository",
                    "accession": "sameAs",
                    "accessible": "isAccessibleForFree",
                    "effectiveOrgan": "keywords",
                    "organ": "keywords",
                    "organPart": "keywords",
                    "disease": "healthCondition.name",
                    "genusSpecies": "species",
                    "lastModifiedDate": "dateModified",
                    "submissionDate": "datePublished",
                },
                "url": "https://www.humancellatlas.org/",
                "conditionsOfAccess": "Varied",
                "genre": "Generalist",
            },
            "flowrepository": {
                "abstract": "Flow Repository is a repository that includes flow cytometry data.",
                "description": "FlowRepository is a database of flow cytometry experiments where you can query and download data collected and annotated according to the MIFlowCyt standard. It is primarily used as a data deposition place for experimental findings published in peer-reviewed journals in the flow cytometry field.",
                "identifier": "Flow Repository",
                "name": "Flow Repository",
                "schema": {
                    "Repository ID": "identifier",
                    "Experiment Name": "name",
                    "Primary researcher": "author.name",
                    "PI/manager": "author.name",
                    "Uploaded by": "author.name",
                    "Experiment dates": "temporalCoverage.startDate,temporalCoverage.endDate",
                    "Dataset uploaded": "datePublished",
                    "Last updated": "dateModified",
                    "Keywords": "keywords",
                    "Manuscripts": "citation,funding",
                    "Organizations": "author.affiliation.name",
                    "Purpose": "description",
                    "Conclusion": "description",
                    "Comments": "description",
                    "Funding": "funding.description",
                    "Quality control": "description",
                },
                "url": "http://flowrepository.org/",
                "conditionsOfAccess": "Open",
                "genre": "Generalist",
            },
            "dash": {
                "abstract": "The Data and Specimen Hub (DASH) is an NICHD supported repository that includes clinical data and specimens.",
                "description": "The NICHD Data and Specimen Hub(DASH) is a centralized resource that allows researchers to share and access de-identified data from studies funded by NICHD. DASH also serves as a portal for requesting biospecimens from selected DASH studies. DASH serves as a mechanism for NICHD-funded extramural and intramural investigators to share research data from studies in accordance with NIH Data Sharing Policies. Many of the NICHD-funded research studies also collected biospecimens that are stored in the NICHD Contracted Biorepository. To provide access to these biospecimens, DASH will store and make available to other investigators the biospecimen catalog for studies that have associated research data in DASH. By supporting data and biospecimen access through DASH, NICHD aims to accelerate scientific findings and improve human health.",
                "identifier": "NICHD DASH",
                "name": "NICHD Data and Specimen Hub (DASH)",
                "schema": {
                    "Study Name": "name",
                    "Study Abbreviation": "alternateName",
                    "citation": "citation.name",
                    "DOI": "citation.doi",
                    "Publication URLs": "citation.url",
                    "NICHD Division/Branch/Center": "funding.funder.name",
                    "Study Description": "description",
                    "Clinical Research Network Name": "author.name",
                    "Principal Investigator(s)": "author.name",
                    "Keywords": "keywords",
                    "Topic": "keywords",
                    "Requires IRB approval to obtain data": "conditionsOfAccess",
                    "Additional Approval Entity": "conditionsOfAccess",
                    "Study Type": "measurementTechnique.name",
                    "ClinicalTrials.gov URL": "mainEntityOfPage",
                    "StudyCollectionStartDate": "temporalCoverage.startDate",
                    "StudyCollectionEndDate": "temporalCoverage.endDate",
                    "approvalDate": "datePublished",
                    "Population Description": "description",
                    "descriptiveDocuments": "hasPart,isBasedOn",
                    "datasetTitle": "name",
                    "datasetDescription": "description",
                    "datasetFormat": "encodingFormat",
                },
                "url": "https://dash.nichd.nih.gov/",
                "conditionsOfAccess": "Restricted",
                "genre": "Generalist",
            },
            "biotools": {
                "abstract": "bio.tools is an ELIXIR supported tool repository that includes information about software tools, databases and services.",
                "description": "The use of bioinformatics is ubiquitous within the life sciences.  In bio.tools, we are striving to provide a comprehensive registry of software and databases, facilitating researchers from across the spectrum of biological and biomedical science to find, understand, utilise and cite the resources they need in their day-to-day work.\nEverything from simple command-line tools and online services, through to databases and complex, multi-functional analysis workflows is included. Resources are described in a rigorous semantics and syntax, providing end-users with the convenience of concise, consistent and therefore comparable information.\nEach bio.tools entry is assigned a human-readable, unique identifier based on the resource name, e.g. biotools:signalp.  These identifiers provide a persistent reference to our "
                "Tool Cards"
                " of essential information, as well as a means to trace resources and integrate bio.tools data with other resources.\nbio.tools was supported by ELIXIR through the  ELIXIR-EXCELERATE grant, which was funded by the European Union Horizon 2020 program under grant agreement 676559.",
                "identifier": "bio.tools",
                "name": "bio.tools",
                "schema": {
                    "name": "name",
                    "description": "description",
                    "homepage": "mainEntityOfPage",
                    "biotoolsID": "identifier, url, _id",
                    "additionDate": "datePublished",
                    "lastUpdate": "dateModified",
                    "topic": "topicCategory",
                    "function": "featureList, input, output",
                    "credit": "author, contributor, funding.funder",
                    "publication": "citation, isBasedOn, citedBy, isBasisFor",
                    "relation": "sameAs, isBasedOn, isBasisFor, hasPart, isPartOf, isRelatedTo",
                    "version": "softwareVersion",
                    "license": "license",
                    "language": "programmingLanguage",
                    "operatingSystem": "operatingSystem",
                    "link": "sdPublisher, codeRepository, isRelatedTo",
                    "download": "softwareHelp, softwareAddOn, downloadUrl, thumbnailUrl, codeRepository, availableOnDevice, applicationSuite",
                    "cost": "isAccessibleForFree",
                    "accessibility": "conditionsOfAccess",
                    "documentation": "softwareHelp",
                    "otherID": "doi",
                    "toolType": "applicationCategory",
                    "collectionID": "keywords",
                    "maturity": "creativeWorkStatus",
                },
                "url": "https://bio.tools/",
                "genre": "Generalist",
                "conditionsOfAccess": "Varied",
                "type": "Computational Tool Repository",
            },
            "figshare": {
                "name": "Figshare",
                "abstract": "Figshare is a NIH supported generalist repository that includes generalist data.",
                "description": "Figshare is a repository where users can make all of their research outputs available in a citable, shareable and discoverable manner.",
                "schema": {
                    "title": "name",
                    "creator": "author",
                    "subject": "keywords",
                    "description": "description",
                    "date": "dateModified",
                    "description": "description",
                    "publisher": "sdPublisher",
                    "type": "@type",
                    "identifier": "doi",
                    "language": "language",
                    "relation": "url",
                    "license": "license",
                    "issued": "datePublished",
                    "sponsor": "funding",
                },
                "url": "https://figshare.com/",
                "identifier": "Figshare",
                "conditionsOfAccess": "Unknown",
                "genre": "Generalist",
            },
        }

        for source, data in source_info.items():
            if source in _meta["src"]:
                _meta["src"][source]["sourceInfo"] = source_info[source]
                _meta["src"][source]["sourceInfo"]["metadata_completeness"] = (
                    self.calculate_metadata_compatibility_average(source)
                )
            elif "parentCollection" in data:
                _meta["src"][source] = {"sourceInfo": source_info[source]}
                _meta["src"][source]["sourceInfo"]["version"] = _meta["src"]["veupath_collections"]["version"]

        return _meta
